src/http_server.rs

```diff
@@ -759,6 +883,26 @@ fn restrict_request(config: &Config, request: ClientRequest) -> ClientRequest {
     }
 }
 
+fn build_request(
```

Comment:
You can see if it's possible to `#[derive(Default)]` for `ClientRequest`. If it is, then you can simply construct it like this:
```rust
ClientRequest {
    query,
    dataset_id,
    dataset_name,
    request_id,
    ..Default::default()
}
```